libqtile/command/base.py

```diff
@@ -207,10 +207,10 @@ def cmd_eval(self, code: str) -> tuple[bool, str | None]:
             error = traceback.format_exc().strip().split("\n")[-1]
             return False, error
 
-    def cmd_function(self, function, *args, **kwargs) -> None:
+    def cmd_function(self, function, *args, **kwargs) -> "Any":
         """Call a function with current object as argument"""
         try:
-            function(self, *args, **kwargs)
+            return function(self, *args, **kwargs)
```

Comment:
I'm interested as to why this is needed. `Drag` has a `start` parameter which takes a lazy function and returns a value so why do we need it for your warp parameter?

Reply:
This took me a bit to figure out so I'll try to explain as best as I can but essentially build-in commands run from a different place, this only affects user defined commands.

All user commands are named `cmd_function`, which is the method I changed, where `function` is the actual user defined function. Meanwhile build-in commands have their own `cmd_<something>` name and they can return whatever they want.

[Here](https://github.com/qtile/qtile/blob/master/libqtile/command/base.py#L147) is the line that chooses the command, `name` is always `function` for user functions which evaluates to the method I changed.

Hope that makes sense

Reply:
I know exactly how they work. I'm just wondering _why_ the change is necessary.

The `start` parameter takes a lazy function but this already works without your change so I'm curious why you need this for your `warp_pointer` argument.

Reply:
It's not specific to `warp_pointer`, it's just so it enables user defined functions to return things.
If you pass the same function as I had in the example to `start` it won't work either without my chanbge

Reply:
But `start` _does_ work without your change from what I can see.

I will test this later though as I may be missing something.

Reply:
Hmm are you sure? I get `TypeError: 'NoneType' object is not subscriptable` in the logs if I give it my function and remove that `return` from `cmd_function`

Reply:
That's why I'm confused.

What I'm trying to say is that, in the default config, Drag has a `start` parameter that takes a lazy function which returns a value. That already works without your change.

Reply:
I was stuck on this too, the difference is that build-in lazy functions get evaluated differently from user-defined functions.

User functions go through `cmd_function` which executes the function but doesn't return anything. Did you try passing a custom function to `start`?

Reply:
Nice. That makes sense. There's no reason not to return the value other than it's not needed 99% of the time!